internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/controller/TestControllerKMERegistrationFromMessageHeader.java

```diff
@@ -0,0 +1,131 @@
+import com.linkedin.venice.integration.utils.VeniceMultiClusterWrapper;
+import com.linkedin.venice.integration.utils.VeniceTwoLayerMultiRegionMultiClusterWrapper;
+import com.linkedin.venice.serialization.avro.AvroProtocolDefinition;
+import com.linkedin.venice.serialization.avro.KafkaValueSerializer;
+import com.linkedin.venice.utils.TestUtils;
+import com.linkedin.venice.utils.Utils;
+import java.util.List;
+import java.util.Optional;
+import java.util.Properties;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.IntStream;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+
+public class TestControllerKMERegistrationFromMessageHeader {
+  private static final Logger LOGGER = LogManager.getLogger(TestControllerKMERegistrationFromMessageHeader.class);
+  private static final int TEST_TIMEOUT = 90_000; // ms
+  private static final int NUMBER_OF_CHILD_DATACENTERS = 1;
+  private static final int NUMBER_OF_CLUSTERS = 1;
+
+  private static final String[] CLUSTER_NAMES =
+      IntStream.range(0, NUMBER_OF_CLUSTERS).mapToObj(i -> "venice-cluster" + i).toArray(String[]::new);
+
+  private List<VeniceMultiClusterWrapper> childDatacenters;
+  private VeniceTwoLayerMultiRegionMultiClusterWrapper multiRegionMultiClusterWrapper;
+
+  @BeforeClass
+  public void setUp() {
+    Properties controllerProps = new Properties();
+    controllerProps.put(DEFAULT_NUMBER_OF_PARTITION_FOR_HYBRID, 2);
+    controllerProps.put(DEFAULT_MAX_NUMBER_OF_PARTITIONS, 3);
+    controllerProps.put(DEFAULT_PARTITION_SIZE, 1024);
+    controllerProps.put(KME_REGISTRATION_FROM_MESSAGE_HEADER_ENABLED, true);
+    multiRegionMultiClusterWrapper = ServiceFactory.getVeniceTwoLayerMultiRegionMultiClusterWrapper(
+        NUMBER_OF_CHILD_DATACENTERS,
+        NUMBER_OF_CLUSTERS,
+        1,
+        3,
+        1,
+        1,
+        1,
+        Optional.of(controllerProps),
+        Optional.of(controllerProps),
+        Optional.empty());
+
+    childDatacenters = multiRegionMultiClusterWrapper.getChildRegions();
+  }
+
+  @AfterClass(alwaysRun = true)
+  public void cleanUp() {
+    Utils.closeQuietlyWithErrorLogged(multiRegionMultiClusterWrapper);
+  }
+
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKMERegistrationThroughAdminTopicChannel() {
+    String clusterName = CLUSTER_NAMES[0];
+    String storeName = Utils.getUniqueString("store");
+
+    VeniceControllerWrapper pController =
+        multiRegionMultiClusterWrapper.getLeaderParentControllerWithRetries(clusterName);
+
+    VeniceMultiClusterWrapper child = childDatacenters.get(0);
+
+    // Remove the latest schema from child controller's local value serializer and remove it from child colo's schema
+    // repository (ZK).
+    VeniceControllerWrapper leaderController = child.getLeaderController(clusterName);
```

Comment:
This is an interesting approach...  But I worry it's too interesting lol.  I wonder, how hard would it be to just read the latest schema and make a doc string change on some field and publish it?  I only ask this as a future proof method of testing this (in case some schema is cached somewhere and the removing/adding post service start up is a NoOp here, but in the production scenario where we add a new schema this may do something else.

Reply:
Yeah, it's a bit involved to develop this testing code. The high level idea here is that I want to create a inconsistent scenario where parent controller has a newer KME version, but a child controller hasn't got it yet. This is to mimic the case where parent controller deploy a new schema before the child and verify that new code can handle it.

With that idea in mind, say if we just update and publish a new schema through e.g. parent controller's `addValueSchema` api, the parent and child controller (through admin messages) are always in sync with their schema and we cannot trigger the new code path anymore. Maybe I misunderstood you comments?

Reply:
I think my concern is that the child controller started up and THEN we removed the zk.  Maybe we bounce the child controller, or, remove the schema from child zk and then start the controller, or, after this step start a new child controller and kill the old one.

Reply:
I think there might be a catch with the proposal. Today, controller, when transitioning from standby->leader, has a logic of loading all schemas from local KafkaMessageEvelop.avsc (`internal/venice-common/src/main/resources/avro/KafkaMessageEnvelope`) file, looping through them and registering in the local cluster, if there is any missing ones in the ZK. So, it means that even if we bounce a child controller after removing it from zk, the newly started controller would always fix it after taking the leadership.

 The current testing approach might be okay, because `HelixReadWriteSchemaRepository` doesn't cache any schemas locally, so even if we remove after the starting up, it doesn't have stale schema.

`KafkaValueSerializer` does cache schemas in memory, thus in the test, we make sure to remove all cached data by `valueSerializer.removeAllSchemas(); `